### ISSUE TITLE
Plugin fails to install if already downloaded

### ISSUE THREAD

**Issue by tech189** [Flow-Launcher/Flow.Launcher#975]
**Describe the bug/issue**
If the plugin zip is already downloaded, installing the plugin will fail. This can happen when python is required for the plugin but it is not found after the user enters its location. When the user puts the correct location of python and tries to install the plugin again, it fails.

**To Reproduce**
1. Install steam plugin
2. Fail to locate python
3. Restart flow launcher
4. Locate python in Settings
5. Install steam plugin

**Your System**
```
Windows build number: 22000.376
Flow Launcher version: 1.9.1
```
**Flow Launcher Error Log**

```
10:49:59.1102+00:00 - ERROR - PluginsManager.InstallOrUpdate - An error occured while downloading plugin
EXCEPTION OCCURS: System.IO.IOException: The file 'C:\Users\me\scoop\apps\flow-launcher\current\app-1.9.1\UserData\Plugins\Steam Search-3.0.4.zip' already exists.
   at SafeFileHandle System.IO.FileStream.ValidateFileHandle(SafeFileHandle fileHandle)
   at SafeFileHandle System.IO.FileStream.CreateFileOpenHandle(FileMode mode, FileShare share, FileOptions options)
   at new System.IO.FileStream(string path, FileMode mode, FileAccess access, FileShare share, int bufferSize, FileOptions options) x 2
   at async Task Flow.Launcher.Infrastructure.Http.Http.DownloadAsync(string url, string filePath, CancellationToken token) in C:/projects/flow-launcher/Flow.Launcher.Infrastructure/Http/Http.cs:line 96
   at async Task Flow.Launcher.Plugin.PluginsManager.PluginsManager.InstallOrUpdate(UserPlugin plugin) in C:/projects/flow-launcher/Plugins/Flow.Launcher.Plugin.PluginsManager/PluginsManager.cs:line 154
```

**Comment by jjw24** [comment]
Hey how's it going?

Could you let me know how you did step 1, did you use pm or directly downloaded the zip into flow's UserData folder?

**Comment by tech189** [comment]
Hey there! 😄 I installed it via the Plugin Store, so using the `pm` command in Flow Launcher.

**Comment by jjw24** [comment]
Did you do it with a fresh install of flow?

The zip shouldnt exist prior to you running pm install via the store.

Let me see if I can replicate.

**Comment by tech189** [comment]
Yes, I think it happened because I messed up the first time picking a python folder so the plugin installation failed midway, leaving the zip there. Once I picked the right folder and deleted the downloaded zip it installed fine!

**Comment by tech189** [comment]
Okay I can't reproduce anymore! It does extract the plugin every time, even if I mess up picking the Python folder and correct it later. So, I'm not sure why it didn't extract it that one time. Maybe you could add a check before starting a new installation to delete any not extracted zips just in case an installation fails midway in the future?

**Comment by jjw24** [comment]
Yeah I can't replicate either.

The plugin download is independent of the python path though, you can download & install first then select python path later.

Unless there was an exception during the download & install process leaving behind the unextracted zip file. Will have another play.

**Comment by tech189** [comment]
>Unless there was an exception during the download & install process leaving behind the unextracted zip file?

Honestly not sure, I can't find any more info in the logs either. Maybe it's worth catching such an exception.

(Btw, it's quite hard to sift through the logs because of hundreds of `Flow.Launcher.Plugin.WindowsSettings.Main.TranslateAllSettings - Resource string for [xxx] not found` errors!)

**Comment by taooceros** [comment]
> >Unless there was an exception during the download & install process leaving behind the unextracted zip file?
> 
> Honestly not sure, I can't find any more info in the logs either. Maybe it's worth catching such an exception.
> 
> (Btw, it's quite hard to sift through the logs because of hundreds of `Flow.Launcher.Plugin.WindowsSettings.Main.TranslateAllSettings - Resource string for [xxx] not found` errors!) 

We are planning to remove that 🤣. That's too much.

**Comment by jjw24** [comment]
Hey @taooceros you have added this to 1.10.0, is it for reducing the warnings from the WindowsSettings plugin?

**Comment by taooceros** [comment]
> Hey @taooceros you have added this to 1.10.0, is it for reducing the warnings from the WindowsSettings plugin?

I think we should fix both either?  I have already finish the resource for windowssetting plugin.

**Comment by VictoriousRaptor** [comment]
Fixed in #2369
